src/main/O2/65430.c

```diff
@@ -0,0 +1,565 @@
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_800663F4.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066428.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066448.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066468.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_800664B0.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_8006671C.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_800667D4.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066894.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_8006692C.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_800669C0.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066A4C.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066AD8.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066D08.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066DE4.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066E88.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066F2C.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80066FA0.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80067114.s")
+
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/func_80067170.s")
+
+void Idle_InitVideo(void);
+void Main_ThreadEntry(void* arg);
+
+extern OSThread sMainThread;
+extern STACK(sMainStack, 0x2700);
+extern StackEntry sMainStackInfo;
+
+extern OSMesgQueue sPiMgrCmdQueue;
+extern OSMesg sPiMgrCmdBuff[8];
+
+#ifdef NON_MATCHING
+// Getting extra nops with the infinite loop, file split?
+void Idle_ThreadEntry(void* arg) {
+    osCreateViManager(OS_PRIORITY_VIMGR);
+    Idle_InitVideo();
+    osViSetSpecialFeatures(OS_VI_DITHER_FILTER_ON | OS_VI_GAMMA_DITHER_OFF | OS_VI_GAMMA_OFF);
+    osViBlack(true);
+    osCreatePiManager(OS_PRIORITY_PIMGR, &sPiMgrCmdQueue, sPiMgrCmdBuff, ARRAY_COUNT(sPiMgrCmdBuff));
+    StackCheck_Init(&sMainStackInfo, sMainStack, STACK_TOP(sMainStack), 0, 0x1000, "main");
+    osCreateThread(&sMainThread, Y_THREAD_ID_MAIN, Main_ThreadEntry, arg, STACK_TOP(sMainStack), Y_PRIORITY_MAIN);
+    D_80108624 = IO_READ(SP_IMEM_START);
+    D_80108628 = IO_READ(SP_DMEM_START);
+    func_8006F500();
+    osStartThread(&sMainThread);
+    osSetThreadPri(NULL, OS_PRIORITY_IDLE);
+
+    for (;;) {}
+}
+#else
+void Idle_ThreadEntry(void* arg);
+#pragma GLOBAL_ASM("asm/us/nonmatchings/main/O2/65430/Idle_ThreadEntry.s")
+#endif
+#undef NON_MATCHING
+
+void Idle_InitVideo(void) {
+    switch (osTvType) {
+        case OS_TV_TYPE_PAL:
+            osViSetMode(&osViModeTable[16]);
+            break;
+
+        case OS_TV_TYPE_NTSC:
+            osViSetMode(&osViModeTable[2]);
+            break;
+
+        case OS_TV_TYPE_MPAL:
+            osViSetMode(&osViModeTable[30]);
+            break;
```

Comment:
I think those indices should be `OS_VI_PAL_LAN1`, `OS_VI_NTSC_LAN1` and `OS_VI_MPAL_LAN1`